File: archive/c/c-plus-plus/remove-all-whitespace.cpp
#include <algorithm>
#include <cctype>
#include <iostream>
#include <ranges>
#include <string>
#include <string_view>

[[noreturn]] void usage() {
    std::cerr << "Usage: please provide a string\n";
    std::exit(1);
}

int main(int argc, char* argv[]) {
    if (argc < 2) usage();

    std::string_view input = argv[1];
    if (input.empty()) usage();

    auto is_not_space = [](unsigned char c) { return !std::isspace(c); };
    auto filtered = input | std::views::filter(is_not_space);

    std::string result;
    std::ranges::copy(filtered, std::back_inserter(result));

    std::cout << result << '\n';
}

File: archive/t/typescript/remove-all-whitespace.ts
function printUsage(): void {
  console.log("Usage: please provide a string");
  process.exit(1);
}

function main(): void {
  const input = process.argv[2];

  if (process.argv.length !== 3 || !input || input.length === 0) {
    printUsage();
  }

  const result = input.replace(/\s/g, "");
  console.log(result);
}

main();
